cmd/kubeadm/app/phases/addons/dns/dns.go

```diff
@@ -272,6 +278,54 @@ func createCoreDNSAddon(deploymentBytes, serviceBytes, configBytes []byte, clien
 	return createDNSService(coreDNSService, serviceBytes, client)
 }
 
+func nodeLocalDNCacheAddon(cfg *kubeadmapi.InitConfiguration, client clientset.Interface) error {
+	nodeLocalIP := "169.254.20.10"
+	if cfg.DNS.NodeLocalDNSCacheIP != "" {
```

Comment:
I'm not sure we should do this kind of defaulting nor to this value, but any defaulting should take place in the API defaulting layer.